tests/components/esphome/test_config_flow.py

```diff
@@ -85,156 +90,195 @@ def __init__(self):
     assert result["data"] == {
-        "host": "127.0.0.1",
-        "port": 6053,
-        "password": "password1",
+        CONF_HOST: "127.0.0.1",
+        CONF_PORT: 6053,
+        CONF_PASSWORD: "password1",
     }
     assert mock_client.password == "password1"
 
 
 async def test_user_invalid_password(hass, mock_api_connection_error, mock_client):
     """Test user step with invalid password."""
-    flow = _setup_flow_handler(hass)
-    await flow.async_step_user(user_input=None)
+    mock_client.device_info = AsyncMock(return_value=MockDeviceInfo(True, "test"))
 
-    mock_client.device_info.return_value = mock_coro(MockDeviceInfo(True, "test"))
+    result = await hass.config_entries.flow.async_init(
+        "esphome",
+        context={"source": "user"},
+        data={CONF_HOST: "127.0.0.1", CONF_PORT: 6053},
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "authenticate"
 
-    await flow.async_step_user(user_input={"host": "127.0.0.1", "port": 6053})
     mock_client.connect.side_effect = mock_api_connection_error
-    result = await flow.async_step_authenticate(user_input={"password": "invalid"})
 
-    assert result["type"] == "form"
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_PASSWORD: "invalid"}
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
     assert result["step_id"] == "authenticate"
     assert result["errors"] == {"base": "invalid_password"}
 
 
 async def test_discovery_initiation(hass, mock_client):
     """Test discovery importing works."""
-    flow = _setup_flow_handler(hass)
+    mock_client.device_info = AsyncMock(return_value=MockDeviceInfo(False, "test8266"))
+
     service_info = {
         "host": "192.168.43.183",
         "port": 6053,
         "hostname": "test8266.local.",
         "properties": {},
     }
+    flow = await hass.config_entries.flow.async_init(
+        "esphome", context={"source": "zeroconf"}, data=service_info
+    )
 
-    mock_client.device_info.return_value = mock_coro(MockDeviceInfo(False, "test8266"))
-
-    result = await flow.async_step_zeroconf(user_input=service_info)
-    assert result["type"] == "form"
-    assert result["step_id"] == "discovery_confirm"
-    assert result["description_placeholders"]["name"] == "test8266"
-    assert flow.context["title_placeholders"]["name"] == "test8266"
+    result = await hass.config_entries.flow.async_configure(
+        flow["flow_id"], user_input={}
+    )
 
-    result = await flow.async_step_discovery_confirm(user_input={})
-    assert result["type"] == "create_entry"
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
     assert result["title"] == "test8266"
-    assert result["data"]["host"] == "test8266.local"
-    assert result["data"]["port"] == 6053
+    assert result["data"][CONF_HOST] == "192.168.43.183"
+    assert result["data"][CONF_PORT] == 6053
+
+    assert result["result"]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5451397)